### PR TITLE
PR Template: Add a (non-enforcing) checklist of items to consider for…

### PR DESCRIPTION
**Pull request by fabiand**
… PRs

**What this PR does / why we need it**:
It introduces a checklist for PR templates.
We want it to help PR authors to take care of all the external dependencies of PRs, such as user-guide, documentation, design.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (dhiller) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: We should add a link to the [reviewer guide "good enough section"](https://github.com/kubevirt/kubevirt/blob/421a6a3acb79204f8fab1e46cb061a33a8f6dbd7/docs/reviewer-guide.md#when-is-a-pr-good-enough) here.

Or we just copy the part into this one.

```suggestion
- [ ] Testing:
  - Unit and e2e tests struck a balance between each other, leaning towards more unit tests and the right approach was chosen
  - Hint: new code requires new unit tests.
  - Hint: new features and bug fixes require at least one e2e test (the core case must be tested).
```

BTW could you elaborate in what you mean with "the right approach was chosen"

Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: > BTW could you elaborate on what you mean with "the right approach was chosen"?

I just tried to appeal at the reader to really think about their decision by including the word *right*. I wanted to them to think twice about what test type to use. But your suggestion is nice, and I'll updat ethe pr - it's much more transparent about what is expected.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (acardace) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: Not every bugfix requires an e2e test, actually most don't, we prefer unit tests :)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (acardace) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
- [ ] Upgrades: Does this change affect upgrades?
```

Or something that refers to changes potentially breaking upgrades.

Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: +1

Comment (dhiller) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: Good point. IIUC then that PR should also be marked with the label `release-note-action-required` and the release note should contain `action required` keywords.

See https://prow.ci.kubevirt.io/command-help#release_note_edit
> PRs that contain 'action required' in their 'release-note' block are given the label 'release-note-action-required' to indicate that the PR introduces potentially breaking changes that necessitate user action before upgrading to the release.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (dhiller) [reply .github/PULL_REQUEST_TEMPLATE.md]: A failing e2e test that proves the bug guards against regressions. However, I agree on that we prefer unit tests, so I'd say let's write many unit tests but few e2e tests.

**Comment by acardace** [comment]
/lgtm
Looks pretty good, thanks @fabiand !

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (dhiller) [inline .github/PULL_REQUEST_TEMPLATE.md:19]:
```suggestion
- [ ] PR: The description will help future contributors
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (dhiller) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion
- [ ] Community: Announcement to the [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered
```

**Review by dhiller — COMMENTED** [review]
nit-picking

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: Meh. Thanks!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md]: Thanks, fixed

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (iholder101) [inline .github/PULL_REQUEST_TEMPLATE.md:19]:
```suggestion
- [ ] PR description: The description will help future contributors
```

File: docs/reviewer-guide.md
Comment (iholder101) [inline docs/reviewer-guide.md:124]:
```suggestion
* New features at least one e2e test (only the core case must be tested)
```

File: docs/reviewer-guide.md
Comment (iholder101) [inline docs/reviewer-guide.md:125]:
```suggestion
* Bug fixes might require an update to existing or the creation of new unit or e2e test
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (iholder101) [inline .github/PULL_REQUEST_TEMPLATE.md:20]:
```suggestion
- [ ] Maintenance: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed. Code should be as clear, easily understandable and simple as possible.
```

**Review by iholder101 — COMMENTED** [review]
Thanks @fabiand! I like this change very much.
Left some small nits, feel free to ignore.

/approve

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: I kept PR:, but enhanced the text, thanks!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (fabiand) [reply .github/PULL_REQUEST_TEMPLATE.md:20]: I referred to KISS and Fowler which is similar

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (iholder101) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: Awesome! Looks much better now IMO
Thanks!

File: docs/reviewer-guide.md
Comment (xpivarc) [inline docs/reviewer-guide.md:124]: What do you mean by `(only the core case must be tested)`?

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: docs/reviewer-guide.md
Comment (fabiand) [reply docs/reviewer-guide.md]: I actually just copied it from the prev version.

I've reworded it.

File: docs/reviewer-guide.md
Comment (0xFelix) [inline docs/reviewer-guide.md:123]: typo: Nw

**Review by 0xFelix — COMMENTED** [review]
Looks good to me, just one nit

File: docs/reviewer-guide.md
Comment (fabiand) [reply docs/reviewer-guide.md]: Aaaaaaaaaaaaaaaargh
the dead of the 10k typos.

Comment (fabiand) [reply docs/reviewer-guide.md]: Thanks

**Comment by fabiand** [comment]
Thanks, fixed

**Review by 0xFelix — APPROVED** [review]
Thanks!

/lgtm

File: docs/reviewer-guide.md
Comment (xpivarc) [inline docs/reviewer-guide.md:124]: This is vague to me. What one test do I need to write? Any, or empty? The previous wording described the reality pretty well. If you going to contribute a new feature you are required to provide tests covering  "core"/basic functionality. Further interesting edge cases can be tested/asked to be tested. We can still emphasize unit tests here, tho.

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: docs/reviewer-guide.md
Comment (fabiand) [reply docs/reviewer-guide.md]: To me the original wording was ... not helpful.

However, I'm restoring the old wording to unblock this pr.

**Comment by fabiand** [comment]
There were different opinions about when e2e tests are needed.
I've restored the old wording, but we migh twant to pickup this topic async in order to see what the right rule of thumb could be.

To me personally `bug fixes require at least on e2e test` is misleading, if there is a code generator bug - no e2e test is required.

**Comment by xpivarc** [comment]
/lgtm
Good PR to see flakes :)

File: docs/reviewer-guide.md
Comment (xpivarc) [reply docs/reviewer-guide.md]: Would you mind elaborating on why you think so? It might be obvious that a new feature needs coverage by e2e tests but I witnessed the opposite.

**Comment by EdDev** [comment]
@AlonaKaplan , FYI.

In order to make sure all are in-sync about this (and agree to enforce it), I think it will be effective to ask a maintainer from each SIG to lgtm such cross-project things in the future.

**Pull request merged**
